quinn-proto/src/endpoint.rs

```diff
@@ -609,6 +623,76 @@ impl Endpoint {
         Ok(())
     }
 
+    /// Reject this incoming connection attempt
+    pub fn reject(&mut self, incoming: Incoming, buf: &mut BytesMut) -> Transmit {
+        self.initial_close(
+            incoming.version,
+            incoming.addresses,
+            &incoming.crypto,
+            &incoming.src_cid,
+            TransportError::CONNECTION_REFUSED(""),
```

Comment:
Is it possible to allow the app specify the reason so  that the upper layer client may do things differently?

Reply:
The spec [specifically warns](https://www.rfc-editor.org/rfc/rfc9000.html#section-10.2.3-3) against this:

> Sending a CONNECTION_CLOSE of type 0x1d in an Initial or Handshake packet could expose application state or be used to alter application state. A CONNECTION_CLOSE of type 0x1d MUST be replaced by a CONNECTION_CLOSE of type 0x1c when sending the frame in Initial or Handshake packets. Otherwise, information about the application state might be revealed. Endpoints MUST clear the value of the Reason Phrase field and SHOULD use the APPLICATION_ERROR code when converting to a CONNECTION_CLOSE of type 0x1c

Reply:
sounds good